plugins/com.google.cloud.tools.eclipse.appengine.newproject.maven/src/com/google/cloud/tools/eclipse/appengine/newproject/maven/MavenAppEngineStandardWizardPage.java

```diff
@@ -75,13 +75,31 @@
   /** True if we're programmatically setting javaPackageField with an auto-generated value */
   private boolean javaPackageProgrammaticUpdate = false;
 
+  private WorkspaceLocationProvider workspaceLocationProvider;
+
+  @VisibleForTesting
+  static interface WorkspaceLocationProvider {
```

Comment:
Why  add an interface here? Seems like overkill. Could we just define a class? For that matter, why a provider instead of just an IPath?